app/assets/javascripts/discourse/app/controllers/delete-topic-confirm.js

```diff
@@ -0,0 +1,38 @@
+import I18n from "I18n";
+import discourseComputed from "discourse-common/utils/decorators";
+import { alias } from "@ember/object/computed";
+import { inject } from "@ember/controller";
+import Controller from "@ember/controller";
+import ModalFunctionality from "discourse/mixins/modal-functionality";
+
+// Modal that displays confirmation text when user deletes a topic
+// The modal will display only if the topic exceeds a certain amount of views
+export default Controller.extend(ModalFunctionality, {
+  topicController: inject("topic"),
+  minViews: alias("siteSettings.min_topic_view_amount_for_delete_confirmation"),
+  deleting: false,
+
+  @discourseComputed("deleting")
```

Comment:
I suggest using the `not` computed property macro here. It's shorter.

Reply:
I'm a bit confused as to how the `not` computed property can be used here, can you please elaborate?

Reply:
My understanding is that you are trying to convert `buttonDisabled` to a boolean. Why do you need it to be a boolean and a truthy / falsey value would not suffice?

Reply:
Ahh, I think I understand what you mean now. I have seen it being used in other components.

Reply:
I ended up deleting the function completely and just accessed the variable directly from the template. `onShow` initializes the default value anyway. The only other function that accesses the `deleting` variable (now changed to `isTopicDeleting`) is `buttonTitle`, which is just a minor UX/UI feature to let the user know that the topic is in the process of being deleted, rather than just a disabled button.